broker/broker.go

```diff
@@ -160,8 +160,10 @@ func (b *Broker) Deprovision(context context.Context, instanceID string, details
 	}
 
 	deployment, err := findDeployment(b.Compose, instanceName)
-	if err != nil {
+	if err == errDeploymentNotFound {
 		return spec, brokerapi.ErrInstanceDoesNotExist
+	} else if err != nil {
```

Comment:
I don't feel strongly about this. I did it this way because we're handling a number of cases about the same variable, and this seemed clearer to me.